React/Views/ScrollView/RCTScrollContentView.m

```diff
@@ -16,15 +16,36 @@ @implementation RCTScrollContentView
 
 - (void)reactSetFrame:(CGRect)frame
 {
-  [super reactSetFrame:frame];
-
 #if !TARGET_OS_OSX // TODO(macOS ISS#2323203)
   RCTScrollView *scrollView = (RCTScrollView *)self.superview.superview;
 #else // [TODO(macOS ISS#2323203)
   // macOS also has a NSClipView in its hierarchy
   RCTScrollView *scrollView = (RCTScrollView *)self.superview.superview.superview;
+
+  if (scrollView != nil) {
+    // On macOS scroll indicators may float over the content view like they do in iOS
+    // or depending on system preferences they may be outside of the content view
+    // which means the clip view will be smaller than the scroll view itself.
+    // In such cases the content view layout must shrink accordingly otherwise
+    // the contents will overflow causing the scroll indicators to appear unnecessarily.
+    NSScrollView *platformScrollView = scrollView.scrollView;
+    CGFloat verticalScrollerWidth = 0;
+    CGFloat horizontalScrollerHeight = 0;
+    if (platformScrollView.scrollerStyle == NSScrollerStyleLegacy) {
+      if (!platformScrollView.verticalScroller.isHidden) {
+        verticalScrollerWidth = platformScrollView.verticalScroller.frame.size.width;
+      }
+      if (!platformScrollView.horizontalScroller.isHidden) {
+        horizontalScrollerHeight = platformScrollView.horizontalScroller.frame.size.height;
```

Comment:
we're getting the size of the scroller from the frame. Can we count on the fact that it has laid out? Or should we be using fittingSize or the like?

Reply:
The autolayout properties like fittingSize and intrinsicContentSize return the same values.

Reply:
But those are not dependent on layout having already happened, which frame is